### ISSUE TITLE
Correct  Places in Gazetteer without Sources

### ISSUE THREAD

**Issue by geoffj-FUG** [FreeUKGen/FreeCENMigration#1337]
When we did the work on the County PARMS and transferred Civil Parishes etc between Counties it created duplicate entries in the Gazetteer for the place. These entries are invalid for Validation purposes but are letting incorrect Places of Birth through as automatically validated. The entries cannot be removed because they have children attached to them. However these entries typically do not have a Source against them and therefore cannot be hidden in the Gazetteer.
We have already proven that it will be a huge task to transfer the children to fix this.
However it appears that these entries typically do not have a Source against them. The entry of a Source is now enforced when a place is added to the Gazetteer. So the entries that are causing the problem appear to be able to be identified by the lack of a source.
The only other entries in the Gazetteer without a Source are where people abused the system to forve validation when we first cut over to CSVProc. It is therefore reasonable to filter out the entries without a Source from the Gazetteer.
Can a filter be set so that any entry without a source is not found in a Gazetteer search, either by the system when validating a piece or by a volunteer when searching the Gazetteer. The only level that needs to see the hidden entries is the data manager.
This will further increasee the integrity of our validation.
Geoff

**Comment by Captainkirkdawson** [comment]
There are 1294 places with no source, of which 474 have data associated with them

**Comment by geoffj-FUG** [comment]
Kirk

 

Can you download me a csv file please and I will have a look at them. Can we
identify the ones with data?

I suspect that the 474 will be the ones that are linked in the PARMS. They
are legitimate to make the PARMS work but we need to hide them so that they
do not get used by the automatic validation process.

There were a lot of entries with no source generated when we started with
the Gazetteer as validators tried to game the system to force things
through. I have disabled a few as they were patently incorrect. But we need
to sort out the issue.

 

Geoff

**Comment by geoffj-FUG** [comment]
An example of a problem entry is Launceston DEV. It was generated when part of the Launceston CON Registration District was moved to DEV to geolocate the Civil Parishes. The result is that Launceston DEV now auto validates,. The incorrect entry can be identified by a lack of a source.
Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG I will get you a list shortly; have been focusing more on the VLD download issue.
I am sorry but i believe your proposed solution will cause as many problems as it fixes.
If a member does not see an entry and the validation does not see an entry then are they not then going to try to create another entry/ This will fail if it has the same name as one that is not being shown; if it is a disabled entry then their attempt at creation will simply reactivate the entry

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG wrote
I think that we can only go about this as a sequential string of tasks.

 

The first is to sort out the Places with no source. That will take me a couple of months allowing for Christmas.

 

Once every valid place has a source we should be able to filter out the places with no source. That will at least filter out the anomalies caused by the PARMS alignment and remove a few incorrect entries like the ABD / FIF sample.

 

Somebody has discovered that they can trigger an acceptance by entering anything into the source field. For instance I found one with a source of 3. I have included several of these with my no source group.

 

I have removed one incorrect AGY Place because it was in the wrong County and the Coordinates were in the sea. I believe that most validators are doing the right thing but there are still a very small number who are not. We can never police everything.

 

I would suggest that once we have a year’s experience we should be able to compile a lookup file that we can use for sources. That will standardise 99% of them. There will always be a need for Other and an explanation of what Other means.

 

There is of course the Elephant in the Room that you have touched on. We need a huge exercise to validate the Places that cannot be linked an entry in the Gazetteer. We will need the ability to enter the alternative POB throughout the database and skip validating a record if the Alternative is acceptable.

 

Working on the database is fraught with danger. I am not sure of the logistics of this but we will eventually need to find a solution.

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG just a reminder we already validate a POB birth place against the alternate names and accept that if found.
We could go further if warranted and do a soundex check of that name against its existence or do a check of the name as part of a gazetteer name ie the options available on the gazetteer search

**Comment by DeniseColbert** [comment]
@rhodamackenzie to send @Captainkirkdawson some examples of duplicate places issue

@Captainkirkdawson to respond to @geoffj-FUG comments

**Comment by Captainkirkdawson** [comment]
Firstly there is the establishment and maintenance of the high quality Gazetteer. Without that the rest of activities involving places become irrelevant.

Secondly there is the matter of ensuring that the Gazetteer is used for all birth places for past, present and future entries. See Ensure that the POB exists within the Gazetteer#1340

Geoff and Rhoda have been working to establish a high quality starting point. So let's address the maintenance of that. FreeREG restricted additions and modifications to a single person as the Place collection was a relatively static one since their prime usage was to reference the location of a register (usually a church). FreeCEN is more dynamic in that regard since it is used to reference what people believed to be their birthplace as interpreted by a 3rd party. Currently editing, creation and deleting are permitted for the following (validator, data_manager, system_administrator, county_coordinator, master_county_coordinator, executive_director, project_manager). Deletion when the place is associated with a PARM must be dealt with by the  system_administrator. However, it would be straightforward to restrict creation of a place to data_manager and system_administrator while permitting the addition of alternate names to a place as at present. It would also be straightforward to produce a monthly report of changes for the data manager to review.

**Comment by PatReynolds** [comment]
Agreed with decision on who to restrict change-making powers to.

@Captainkirkdawson to implement

**Comment by PatReynolds** [comment]
@geoffj-FUG still working on this, then @Captainkirkdawson to make changes

**Comment by geoffj-FUG** [comment]
Message sent to Kirk with first cut spreadsheet attached:

Kirk

I have attached my first cut revised version of the Gazetteer entries.

Warning – do not change the filters on this copy or you will never get them back. They were quite complex in the end.

The entries are in three colours.
Black – these are entries that I have fixed up so far and need no further action.
Green – These are entries that are in the wrong County or are duplicate places. They are all linked to the PARMS and have no source entered. They need verifying as caused by the PARMS exercise. They should then either be relinked to the correct place and disabled or can be left as no source and filtered out  of the Gazetteer for validation purposes but left in with no filter for the PARMS links.
Red – These are first pass entries that have not been resolved. My next step is to revisit each of these and see what needs to be done to resolve them. 

I am concerned at the number of source entries that have no url. We need to make the url entry compulsory as well. I have managed to attach a url to every genuine and correct entry that I have amended so far. Looking at entries with no url is another issue that will have to be addressed later.

Can you extract a list of unique sources please? Someone in YKS is putting in the source as the ‘Gazetteer’ instead of ‘Genuki’. I now need to go through all of these to identify them. I would like you to amend every ‘Gazetteer’ entry with an url to ‘Gazetteer of British Place Names’ which is the correct reference acknowledgement and then I can spot the incorrect entries. Whoever is doing it is not entering urls.

Once we have a list of unique Sources I would like to convert the field to a lookup field so that we have a standardised source name. It is currently a list with mis-spellings and abbreviations in it. I have fixed up several as I included them in my filters. We will end up with a category of other but if we make urls compulsory that will become transparent.

Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG I need clarification. See the specific stories for your action requests.
Amend every ‘Gazetteer’ entry with an url to ‘Gazetteer of British Place Names’ #1360
Extract a list of unique values in the sources field of the Gazetteer collection #1359

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG was I expected to take any action on the content e.g. deleting entries having made adjustments to linkages?

**Comment by geoffj-FUG** [comment]
Kirk
I can go no further with the entries in green. They all appear to be as a result of the PARMS exercise. Many of them have been identified as duplicate entries. If we re-link them to the main entry we can then remove the duplicate.
From my point of view the entry from the Gazetteer of British Place Names has been the primary one.
Once we have done that we should be able to set a filter for validation that does not use entries with no source. That will clean up that aspect of the problem.
I still need to go back through the entries in red. I may be able to resolve some of them or at least expand on the notes that I have provided.
Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG so to be totally clear (The statement above is a reason for action) you wish me to resolve those entries that are coloured green? If that is correct I will create the appropriate story for action by by me.

**Comment by geoffj-FUG** [comment]
Kirk
We need to resolve the entries that are coloured green. They all appear to be associated with the PARMS.
There will be 2 solutions:
1. Where the place is a duplicate or where the correct place name exists the PARMS will need to be linked to the correct entry. Once this is done the problem place name in the Gazetteer need to be disabled.
2. Where the place name is verified as correct the PARMS entry will be able to be left as is. Where there is no source it is usually because the source is in a different County and the entry was as a result of the PARMS moves. Once all of these are identified and all sources are entered the Gazetteer entries with no Sources will be able to be filtered out for all needs except PARMS needs.
This will clean up those particular identified PARMS related entries,.
Geoff

**Comment by geoffj-FUG** [comment]
Text of email sent to Kirk with spreadsheet attached:

I have attached the next version of the Gazetteer amendments. Some of the previous red entries have been resolved and are now black as no action is required.

The spreadsheet is colour coded as follows:

Green – These are the same list as was included in green on the first version of the spreadsheet. None have been changed
Blue – These are entries that were in red for me to look at, but which have now been determined as apparently due to the PARMS exercise. The instructions for these are exactly the same as for the first set of green entries.
Red – These are problem entries that cannot be disabled because search records are attached to them. The search records need to be attached to the correct Place Name and the entry that is in red disabled. The correct Place names have been identified. 
Some red entries are duplicates where the Place Name appears twice.
There are a few entries that I cannot identify the exact correct Place Name. I have suggested that these need to be referred to the County Coordinator for advice. They will no doubt need the piece number that the search records are attached to so that they can investigate the entry.

When all this is done can you send me a new copy of the master spreadsheet with web sites included as well please?

Geoff

**Comment by PatReynolds** [comment]
@rhodamackenzie and @geoffj-FUG to work at this list.

**Comment by geoffj-FUG** [comment]
Pat

 

Kirk is working on this at the moment. It cannot be done from a PC because there are links that need removing and reassigning.

 

Kirk is currently running a rake task to do it. It is taking about 15 minutes per entry. It is a slow process but it is happening.

 

Geoff

**Comment by PatReynolds** [comment]
Sorry, @geoffj-FUG I obviously minunderstood what I should have noted.

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG I have completed the red blue and green changes in their respective stories. I have retuned the listing to you for information

Further changes arising from the requests to coordinators should first be referred to @rhodamackenzie who has the ability to make almost all changes.

Only changes that require the merging of place entries because they have the same location (and are NOT alternates) should come to me as they can only be addressed on an individual case basis by a rake task

**Comment by geoffj-FUG** [comment]
Kirk

 

Thank you.

 

I will break up the spreadsheet that I have into the two parts.

 

We are almost there at last!

 

Geoff

**Comment by PatReynolds** [comment]
all remianing actions on this story are now part of #1408, so closing.
